scripts/ci/pre_commit/compile_ui_assets.py

```diff
@@ -68,10 +69,14 @@ def get_directory_hash(directory: Path, skip_path_regexp: str | None = None) ->
         shutil.rmtree(dist_directory, ignore_errors=True)
     env = os.environ.copy()
     env["FORCE_COLOR"] = "true"
-    subprocess.check_call(
-        ["pnpm", "install", "--frozen-lockfile", "--config.confirmModulesPurge=false"],
-        cwd=os.fspath(ui_directory),
-    )
+    if os.getenv("AIRFLOW_PRE_CACHED_PNPM_PACKAGES", "false") == "true":
+        # Copy pnpm-cache to node_modules from pnpm-cache
+        shutil.copytree(pnpm_node_modules_cache_directory, node_modules_directory, dirs_exist_ok=True)
+    else:
```

Comment:
One small tthing here. I think we should ALWAYS run the install step below. This covers case where the .lock file changed in the meantime and we just need to update it based on pre-cached node_modules